### DISCUSSION TITLE
time value of simulation

### DISCUSSION THREAD

**Discussion by Mric237** [thliebig/openEMS-Project#127] [Q&A Usage]
Hi i am new in openems and am trying to do some simple test with openems like sending a plane wave excitation from a specific source and measure it at different position. I am trying also reflexion with sphere and plate. Now the first problem is to get the time of the simulation in python so that i can plot the field dependant on time. I am measuring the field with a dumb box and get the values as h5. file.  please how can i do it?

**Comment by thliebig** [comment]
It's bin a while that I looked at the hdf5 files but it should contain all the information required. Only make sure that the TD (time domain) data are stored, because I think it is not the default?
The hdf5 file should contain the time-step and the data should state the timestep used for every field data.

Maybe you can have a look at the matlab functions for reading the hdf5 dumps. We truly need something similar for python...

**Reply by Mric237** [reply]
Thank you. In Matlab it is okay. I thought it was also possible in python. My second problem would be to measure the field reflected from a plate or sphere. But the dump box should not receive field from the source. I have try to put the dump box behind a pec plate and leave some space so that the sphere could receive some field. But I think i have some diffraction under the plate. May be do you have some ideas for me please?

**Reply by thliebig** [reply]
Sorry but I have not enough details to answer your question about the reflection etc. 
I think you should create a useful planar dump-box and observe how the fields evolve over time to get some insight into what works and what doesn't as you expect it.

**Reply by Mric237** [reply]
sorry for bothering again, but i have too many questions about the solver. In the attached image, you can see a source, three dump boxes over the source and a dump box in front of it. I want to know how to send an E-field in y direction with the propagation in z direction. So the three dump boxes could not measure this field. My first objective is to replace the dump box in front of the source with an cylinder(PEC) that will reflect the field on the three dump boxes.  Later the three dump boxes could be moved on the side to have a reflection in x direction.


![image](https://github.com/thliebig/openEMS-Project/assets/146964061/0bd681f2-1a95-4f80-bbef-f94cc0bedd0e)
